### PR TITLE
TST: Add GIFTI tests

### PR DESCRIPTION
**Pull request by bcipolli**
This isn't complete, but add some basic smoke tests (at least `print_summary` shouldn't raise an error), that explicit errors are hit, and that non-`None` defaults are set by objects.

### CODE REVIEW

**Comment by bcipolli** [comment]
@effigies besides the other two, I also rebased this one. Anything you'd like to see here? It's just adding a bit to test coverage.

File: nibabel/gifti/gifti.py [hunk 1]
```diff
@@ -453,7 +453,7 @@ def add_gifti_data_array(self, dataarr):
         if isinstance(dataarr, GiftiDataArray):
             self.darrays.append(dataarr)
         else:
-            print("dataarr paramater must be of tzpe GiftiDataArray")
+            raise ValueError("dataarr paramater must be of type GiftiDataArray")
```
Comment (effigies) [inline nibabel/gifti/gifti.py]: Actually, while I'm critiquing, my preference on this idiom would be:

    if not isinstance(...):
        raise TypeError(...)
    self.darrays.append(dataarr)

Just associates the check more closely with the guarded case, and avoids visually demoting the useful code by putting it in an `else` block.

**Comment by bcipolli** [comment]
@effigies I'm seeing an AppVeyor failure in this PR. I tried to follow it backwards to previous PRs, but none completed it. I also didn't reproduce on my local config. What's AppVeyor? And do we care? Previous PRs were merged without AppVeyor finishing.

File: nibabel/gifti/gifti.py (hunk 1 above)
Comment (bcipolli) [inline nibabel/gifti/gifti.py]: Good call on the style and error type. Fixed, and scrubbed the code for other `ValueError`s that should be `TypeError`s. Tests running.

**Comment by effigies** [comment]
It does Windows builds, at least for nibabel. I left a note on #358. 32 bit builds are broken under numpy 1.10, I believe for floating point precision issues.

That's beyond my skill/time to resolve, so as long as that's the only error, I'm comfortable merging.

On October 17, 2015 8:33:16 AM EDT, Ben Cipollini <notifications@github.com> wrote:
>@effigies I'm seeing an AppVeyor failure in this PR. I tried to follow
>it backwards to previous PRs, but none completed it. I also didn't
>reproduce on my local config. What's AppVeyor? And do we care? Previous
>PRs were merged without AppVeyor finishing.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/nipy/nibabel/pull/355#issuecomment-148913702

-- 
Chris Markiewicz

File: nibabel/gifti/gifti.py [hunk 2]
```diff
@@ -450,10 +451,9 @@ def add_gifti_data_array(self, dataarr):
         ----------
         dataarr : GiftiDataArray
         """
-        if isinstance(dataarr, GiftiDataArray):
-            self.darrays.append(dataarr)
-        else:
-            print("dataarr paramater must be of tzpe GiftiDataArray")
+        if not isinstance(dataarr, GiftiDataArray):
+            raise TypeError("dataarr paramater must be of type GiftiDataArray")
```
Comment (effigies) [inline nibabel/gifti/gifti.py]: For consistency, how about "Not a valid GiftiDataArray instance"?

File: nibabel/gifti/gifti.py [hunk 3]
```diff
@@ -404,7 +405,7 @@ def labeltable(self, labeltable):
 
         """
         if not isinstance(labeltable, GiftiLabelTable):
-            raise ValueError("Not a valid GiftiLabelTable instance")
+            raise TypeError("Not a valid GiftiLabelTable instance")
```
Comment (effigies) [inline nibabel/gifti/gifti.py]: Let's do a test for this exception, while we're at it.

**Comment by effigies** [comment]
Noting for posterity that although these `ValueError => TypeError` changes are API changes, the exceptions were introduced in the previous PR.
